src/widgets/signature/SignatureWidget.js

```diff
@@ -43,70 +41,55 @@ const ClearButton = styled(IconButton)`
 `;
 
 type Props = {
-  disabled :boolean,
-  value :string,
-  onChange :(data :string | void) => void
+  disabled :boolean;
+  onChange :(data :string | void) => void;
+  value :?string;
 };
 
-class SignatureWidget extends Component<Props> {
-
-  signatureCanvasRef :{ current :?SignatureCanvas } = React.createRef();
-
-  componentDidUpdate(prevProps :Props) {
-    const { disabled, value } = this.props;
-    const { current } = this.signatureCanvasRef;
-    if (disabled && current) {
-      current.off();
-    }
-    else if (current) {
-      current.on();
-    }
+const SignatureWidget = (props :Props) => {
 
-    if (isNonEmptyString(value)) {
-      const newCanvasValue :string = tryConvertingToDataURL(value);
-      const prevCanvasValue :string = tryConvertingToDataURL(prevProps.value);
-      if (newCanvasValue !== prevCanvasValue) {
-        this.setCanvas(newCanvasValue);
-      }
-    }
-  }
+  const { disabled, onChange, value } = props;
+  const signatureCanvasRef :{ current :?SignatureCanvas } = useRef();
 
-  handleClear = () => {
-    const { onChange } = this.props;
-    const { current } = this.signatureCanvasRef;
+  useEffect(() => {
+    const { current } = signatureCanvasRef;
     if (current) {
-      current.clear();
-      onChange();
+      if (disabled) {
+        current.off();
+      }
+      else {
+        current.on();
+      }
+      if (value) {
+        const valueAsDataURL = tryConvertingToDataURL(value);
+        current.fromDataURL(valueAsDataURL);
+      }
     }
-  }
+  }, [disabled, signatureCanvasRef, value]);
```

Comment:
It's not a big deal, but ideally the logic to toggle the disabled and the dataURL conversion would have their own separate hooks since they're not related to each other. If this were a heavier component, you wouldn't want to always convertDataURL whenever the canvas is enabled/disabled. But since toggling the disabled state is not something the user controls and should only happen once, this is kind of a non-issue.

Just a general tip.